### PR TITLE
Store the date of the first day of new employees

### PR DESCRIPTION
**Pull request by Hellikrox**
# Please make sure all below checkboxes in the Mandatory section have been checked before submitting your PR (also don't forget to pay attention to the Depending on Circumstances section)

## Mandatory

- [x] The commit message is an imperative and starts with a capital letter (for example, `Add something`, `Remove something`, `Fix something`, etc.).
- [x] I made sure I hadn't put a period at the end of the the commit message(s).

## Depending on Circumstances

Some of the items in the section become mandatory if you are a first-time contributor and/or your changes are relevant to the items.

- [ ] Adding a new parameter I didn't forget to reflect it in `README.md`.
- [ ] Modifying a list of constants or imports I didn't brake an alphabetical order.
- [x] (for first-time contributors) One of the commits in the pull request adds me to the **Acknowledgements** section in `AUTHORS.md`.

### CODE REVIEW

File: src/birthder.js [hunk 1]
```diff
@@ -421,6 +425,10 @@
         if (!user.dateOfBirth) {
           robot.adapter.sendDirect({ user: { name: user.name } }, `Welcome to ${COMPANY_NAME}! :tada:\nEmm... where was I?\nOh! Please, enter your date birth (DD.MM.YYYY).`)
         }
+        const today = moment().format('DD.MM.YYYY')
```
Comment (eugulixes) [inline src/birthder.js]: You should use `OUTPUT_DATE_FORMAT` instead of `DD.MM.YYYY`.

File: src/birthder.js [hunk 2]
```diff
@@ -573,6 +581,76 @@
...
+          const sortedDate = date.unix() >= moment().unix()
+            ? date.format(`DD.MM.${date.year() - 1}`) : date.format(`DD.MM.${date.year()}`)
+
+          return {
+            name: user.name,
+            dateOfStart: user.dateOfStart,
+            sortedDate: sortedDate
+          }
+        })
+        .sort((a, b) => sorting(a.sortedDate, b.sortedDate, 'DD.MM.YYYY'))
```
Comment (eugulixes) [inline src/birthder.js]: There should be `OUTPUT_DATE_FORMAT` instead of `DD.MM.YYYY` as well.

File: src/birthder.js [hunk 3]
```diff
@@ -573,6 +581,76 @@
       msg.send(`*Birthdays list*\n${message.join('\n')}`)
     })
 
+    //Reset date of first working day
+    robot.respond(routes.start, async (msg) => {
```
Comment (eugulixes) [inline src/birthder.js]: The command must be restricted only to admins.

File: src/birthder.js [hunk 4]
```diff
@@ -9,6 +9,8 @@
 //   hubot birthday set <username> <date>.<month>.<year> - sets a birthday for the user (privileged: admins only)
 //   hubot birthdays on <date>.<month> - shows a list of users with a set birthday date
 //   hubot birthday delete <username> - deletes birthday for the user (privileged: admins only)
+//   hubot firstday set <username> <date>.<month>.<year> - sets a first working day for the user
```
Comment (eugulixes) [inline src/birthder.js]: `... (privileged: admins only)`

File: src/birthder.js [hunk 5]
```diff
@@ -421,6 +425,9 @@
         if (!user.dateOfBirth) {
           robot.adapter.sendDirect({ user: { name: user.name } }, `Welcome to ${COMPANY_NAME}! :tada:\nEmm... where was I?\nOh! Please, enter your date birth (DD.MM.YYYY).`)
         }
+        const today = moment().format(OUTPUT_DATE_FORMAT)
+        user.dateOfStart = today
+        robot.adapter.sendDirect({ user: { name: user.name } }, `Congrats! Today is your first day in here. :clap:`)
```
Comment (eugulixes) [inline src/birthder.js]: I guess the message is redundant.

File: src/birthder.js [hunk 6]
```diff
@@ -573,6 +580,81 @@
...
+        return msg.send(`Saving ${name}'s first working day.`)
+      } else if (users.length > 1) {
+        return msg.send(getAmbiguousUserText(users))
+      } else {
+        return msg.send(`I have never met ${name}.`)
+      }
+    })
+
+    // Print sorted users first working days.
+    robot.respond(routes.slist, async function (msg) {
```
Comment (eugulixes) [inline src/birthder.js]: Let's put it simple: `async (msg) =>`

File: src/birthder.js [hunk 7]
```diff
@@ -573,6 +580,81 @@
...
+            ? date.format(`DD.MM.${date.year() - 1}`) : date.format(`DD.MM.${date.year()}`)
+
+          return {
+            name: user.name,
+            dateOfStart: user.dateOfStart,
+            sortedDate: sortedDate
+          }
+        })
+        .sort((a, b) => sorting(a.sortedDate, b.sortedDate, OUTPUT_DATE_FORMAT))
+        .map(user => ` @${user.name} join our team ${moment(user.dateOfStart, DATE_FORMAT).format(OUTPUT_DATE_FORMAT)}`)
```
Comment (polina-popova) [inline src/birthder.js]: You mostly copy the filtering from `birthday list` command, may be better will be not repeat but reuse this piece of code and strike it out as a function.

File: src/birthder.js [hunk 8]
```diff
@@ -537,8 +542,21 @@
...
+      let attr, desc, title
+
+      if (msg.match[1] === 'birthdays') {
+        attr = 'dateOfBirth'
+        desc = `was born on`
+        title = `Birthdays`
+      }
+      if (msg.match[1] === 'fwd') {
+        attr = 'dateOfFwd'
+        desc = `join our team`
```
Comment (polina-popova) [inline src/birthder.js]: `joined our team`

**Review by polina-popova — COMMENTED** [review]
Looks good it is just one small grammar fix.

**Review by polina-popova — APPROVED** [review]

**Pull request merged**